main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+  "context"
+  "log"
+  "fmt"
+  "time"
+  "github.com/andrecrts/go-bootcamp/infrastructure/router"
+  "github.com/andrecrts/go-bootcamp/domain/model"
+	"github.com/labstack/echo/v4"
+
+  "go.mongodb.org/mongo-driver/bson"
+  "go.mongodb.org/mongo-driver/bson/primitive"
+   "go.mongodb.org/mongo-driver/mongo"
+   "go.mongodb.org/mongo-driver/mongo/options"
+   // "go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+
+func insertProduct(client *mongo.Client, productId string, name string ) {
+  product := model.Product{productId, name}
+
+  collection := client.Database("bootcamp").Collection("products")
+
+  insertResult, err := collection.InsertOne(context.TODO(), product)
+
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println("Inserted post with ID:", insertResult.InsertedID)
+}
+
+
+func getPost(client *mongo.Client, id primitive.ObjectID) {
+
+  collection := client.Database("bootcamp").Collection("products")
+
+  filter := bson.D{}
+
+  var product model.Product
+
+  err := collection.FindOne(context.TODO(), filter).Decode(&product)
+
+  if err != nil {
+
+  log.Fatal(err)
+
+  }
+
+
+
+  fmt.Println("Found product with name ", product.Name)
+
+}
+
+func handleRequests() {
+
+  client, err := mongo.NewClient(options.Client().ApplyURI("mongodb://localhost:27017"))
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+  err = client.Connect(ctx)
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  defer client.Disconnect(ctx)
+
+  // insertProduct(client, "Test", "123")
+  oid, err := primitive.ObjectIDFromHex("5fb760418d726a2670df8b88")
+  getPost(client, oid)
+
+  e := echo.New()
+  e = router.NewRouter(e)
+
+	if err := e.Start(":1000"); err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func main() {
+  handleRequests()
```

Comment:
Although this is not bad, I'd suggest deep dive into what software architecture is.

In the main method you should be creating instances that you would use later on, and prepare them to be able to communicate between them. Send them the required data to be able to perform their responsibilities.

Specifically speaking about this Bootcamp, we are trying to evangelize everyone on clean architecture (you are free to use whatever you want though); I'll leave you these posts that might help you with that:

https://eltonminetto.dev/en/post/2018-03-05-clean-architecture-using-go/
https://eltonminetto.net/en/post/2020-07-06-clean-architecture-2years-later/